packages/devextreme/js/__internal/ui/tree_view/m_tree_view.search.ts

```diff
@@ -29,50 +96,60 @@ const TreeViewSearch = TreeViewBase.inherit(searchBoxMixin).inherit({
         this._repaintContainer();
         this.option('focusedElement', null);
         break;
-      case 'searchExpr':
+      }
+      case 'searchExpr': {
         this._initDataAdapter();
         this.repaint();
         break;
-      case 'searchMode':
-        this.option('expandNodesRecursive') ? this._updateDataAdapter() : this._initDataAdapter();
+      }
+      case 'searchMode': {
+        const { expandNodesRecursive } = this.option();
+        if (expandNodesRecursive) {
+          this._updateDataAdapter();
+        } else {
+          this._initDataAdapter();
+        }
         this.repaint();
         break;
+      }
+      case 'searchTimeout':
+        break;
       default:
-        this.callBase(args);
+        super._optionChanged(args);
     }
-  },
+  }
 
-  _updateDataAdapter() {
+  _updateDataAdapter(): void {
     this._setOptionWithoutOptionChange('expandNodesRecursive', false);
 
     this._initDataAdapter();
 
     this._setOptionWithoutOptionChange('expandNodesRecursive', true);
-  },
+  }
 
   _getDataAdapterOptions(): Partial<DataAdapterOptions> {
+    const { searchValue, searchMode, searchExpr } = this.option();
     return {
-      ...this.callBase(),
-      searchValue: this.option('searchValue'),
-      searchMode: this.option('searchMode') || 'contains',
-      searchExpr: this.option('searchExpr'),
+      ...super._getDataAdapterOptions(),
+      searchValue,
+      searchMode: searchMode ?? 'contains',
+      // @ts-expect-error ts-error
```

Comment:
Let's specify a concrete type for searchExpr and extend it according to public expectations
```
searchExpr?: string | Function | (string | Function)[] | ((item: ItemData) => unknown);
```